src/main/java/com/codingnomads/betty/logic/services/ExpectedValueCalculationService.java

```diff
@@ -0,0 +1,41 @@
+package com.codingnomads.betty.logic.services;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Service
+public class ExpectedValueCalculationService {
+
+    private AnalyzeTweetsService analyzeTweetsService;
+    private GameInformationService gameInformationService;
+
+    @Autowired
+    public ExpectedValueCalculationService(AnalyzeTweetsService analyzeTweetsService, GameInformationService gameInformationService) {
+        this.analyzeTweetsService = analyzeTweetsService;
+        this.gameInformationService = gameInformationService;
+    }
+
+    public Map<String, Double> calculateExpectedValue(String homeTeamName, String awayTeamName){
+
+        Map<String, Double> expectedValueMap = new HashMap<>();
+
+        double probabilityWithTweets = analyzeTweetsService.calculateProbabilityWithTweets(homeTeamName).getProbabilityToWin();
+
+        Map<String, Double> marketOdds = gameInformationService.getOddsByMatch(homeTeamName, awayTeamName);
+
+        double expectedValue = probabilityWithTweets * (marketOdds.get(homeTeamName) - 1) - (1 - probabilityWithTweets) * marketOdds.get(awayTeamName);
```

Comment:
Shouldn't this be double expectedValue = probabilityWithTweets * (marketOdds.get(homeTeamName) - 1) - (1 - probabilityWithTweets) * **(**(marketOdds.get(awayTeamName) **- 1)** ???

Also this will only hold true for back bets, not lay bets.

Reply:
actually ignore my comment about the -1 in bold. Its is wrong. The formula still needs editing for lay betting though